examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,391 @@
+        raise ValueError("No topic selected.")
+
+    topic_name, pro_key, con_key = selected_topic_tuple
+    side = select_side(topic_name)
+    return topic_name, pro_key, con_key, side
+
+
+def is_llm_delegate() -> bool:
+    """Prompt the user to decide on LLM delegation.
+
+    Asks the user whether the LLM should autonomously continue the debate
+    without requiring user input.
+
+    Returns:
+        bool: True if the user chooses LLM delegation, otherwise return False.
+    """
+    return Confirm.ask(
+        "Would you like the LLM to autonomously continue the debate without "
+        "waiting for user input? or ask for your input first time?",
+        default=False,
+    )
+
+
+def is_same_llm_for_all_agents() -> bool:
+    """Prompt the user to decide if same LLM should be used for all agents.
+
+       Asks the user whether the same LLM should be configured for all agents.
+
+       Returns:
+           bool: True if the user chooses same LLM for all agents, otherwise return False.
+       """
+    # Ask the user if they want to use the same LLM configuration for all agents
+    return Confirm.ask(
+        "Do you want to use the same LLM for all agents?",
+        default=True,
+    )
+
+
+def is_google_api_key_configured() -> bool:
+    """Prompt the user to validate if the Google API Key and CSE ID are configured.
+
+    Asks the user whether they have configured the Google API key and CSE ID.
+
+    Returns:
+        bool: True if the user has configured the required env variables for Google API Key and CSE ID, otherwise
+        return False.
+    """
+    return Confirm.ask(
+        "Do you have a Google API key and configured? For example, GOOGLE_API_KEY=your-key and GOOGLE_CSE_ID=your-cse-id",
+        default=False,
+    )
+
+
+def run_debate() -> None:
+    """Execute the main debate logic.
+
+    Orchestrates the debate process, including setup, user input, LLM agent
+    interactions, and final feedback. Handles both user-guided and LLM-
+    delegated debates.
+
+    This function:
+    1. Loads global settings and the base LLM configurations.
+    2. Prompts the user to confirm if they want to use same LLM for all agents.
+    3. Prompts the user to select a debate topic and a side(Pro or Con).
+    4. Sets up pro, con, and feedback agents.
+    5. Runs the debate for a specified number of turns, either interactively
+       or autonomously.
+    6. Provides a feedback summary at the end.
+    7. Optionally validates references if a Google API key is configured. Creates a references validation agent.
+    Utilizes the langroid GoogleSearchTool.
+    """
+
+    global_settings = get_global_settings(nocache=True)
+    lr.utils.configuration.set_global(global_settings)
+    same_llm: bool = is_same_llm_for_all_agents()
+    llm_delegate: bool = is_llm_delegate()
+
+    # Get base LLM configuration
+    if same_llm:
+        shared_agent_config: OpenAIGPTConfig = get_base_llm_config("for main LLM")
```

Comment:
I think the word "for" should be dropped since the template already has "for" in it